### PR TITLE
Validate URLs within the admin, fix #277

### LINKED ISSUE #277: Validate URLs within the admin
As an administrator, I want to be notified when I submit an incorrect URL in order to avoid confusion for users trying to download a resource.

### PR DESCRIPTION
**Pull request by davidbgk**

### CODE REVIEW

**Comment by vinyll** [comment]
I'm not familiar enough with _Vue_ to review the concept but it all makes sense to me.

File: js/components/form/base-form.js [hunk 1]
```diff
@@ -136,7 +136,7 @@ export default {
                 }
             },
             errorPlacement: function(error, element) {
-                $(element).closest('.form-group,.field-wrapper').append(error);
+                $(element).closest('.form-group,.field-wrapper').children('div').append(error);
```
Comment (noirbizarre) [inline js/components/form/base-form.js]: Are you sure ? I think the extra `<div/>` breaks the bootstrap style: http://getbootstrap.com/css/#forms-control-validation

File: js/components/form/url-field.vue [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<template>
+<div>
+    <input type="text" class="form-control" v-el:url-field
```
Comment (noirbizarre) [inline js/components/form/url-field.vue]: Using html 5 `type="url"` should automatically add the validation rule

File: js/components/form/url-field.vue [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import {FieldComponentMixin} from 'components/form/base-field';
+
+export default {
+    name: 'url-field',
+    replace: true,
+    mixins: [FieldComponentMixin],
+    events: {
+        'form:ready'() {
+            // Perform all validations on end field because performing on start field unhighlight.
+            $(this.$els.urlField).rules('add', {
```
Comment (noirbizarre) [inline js/components/form/url-field.vue]: not needed if using `type=url"`

**Comment by noirbizarre** [comment]
r+wc

File: js/components/form/base-form.js (hunk 1 above)
Comment (davidbgk) [inline js/components/form/base-form.js]: Actually, I'm not creating a div but looking for it? Visually it looks better.

File: js/components/form/url-field.vue (hunk 2 above)
Comment (davidbgk) [inline js/components/form/url-field.vue]: Oh didn't think about that possibility. Clever.

File: js/components/form/base-form.js (hunk 1 above)
Comment (noirbizarre) [inline js/components/form/base-form.js]: OK, does it work with both horizontal and vertical form ? If yes, it's fine to me.

Comment (davidbgk) [inline js/components/form/base-form.js]: I confirm that it renders correctly:

capture d ecran 2015-12-09 a 16 35 40

capture d ecran 2015-12-09 a 16 35 47

File: js/components/form/url-field.vue (hunk 3 above)
Comment (davidbgk) [inline js/components/form/url-field.vue]: It works!

**Pull request merged**